src/lib/markbind/src/parser.js

```diff
@@ -581,53 +570,52 @@ class Parser {
     });
   }
 
-  _rebaseReference(node, foundBase) {
+  /**
+   * Pre-renders the baseUrl of the provided node and its children according to
+   * the site they belong to, such that the final call to render baseUrl correctly
+   * resolves baseUrl according to the said site.
+   */
+  _rebaseReference(node) {
     if (_.isArray(node)) {
-      return node.map(el => this._rebaseReference(el, foundBase));
+      return node.map(el => this._rebaseReference(el));
     }
     if (Parser.isText(node)) {
       return node;
     }
-    // Rebase children element
-    const childrenBase = {};
-    node.children.forEach((el) => {
-      this._rebaseReference(el, childrenBase);
-    });
-    // rebase current element
-    if (node.attribs[ATTRIB_INCLUDE_PATH]) {
-      const filePath = node.attribs[ATTRIB_INCLUDE_PATH];
-      let newBaseUrl = urlUtils.calculateNewBaseUrls(filePath, this.rootPath, this.baseUrlMap);
-      if (newBaseUrl) {
-        const { relative, parent } = newBaseUrl;
-        // eslint-disable-next-line no-param-reassign
-        foundBase[parent] = relative;
-      }
-      // override with parent's base
-      const combinedBases = { ...childrenBase, ...foundBase };
-      const bases = Object.keys(combinedBases);
-      if (bases.length !== 0) {
-        // need to rebase
-        newBaseUrl = combinedBases[bases[0]];
-        if (node.children) {
-          // ATTRIB_CWF is where the element was preprocessed
-          const currentBase = urlUtils.calculateNewBaseUrls(node.attribs[ATTRIB_CWF],
-                                                            this.rootPath, this.baseUrlMap);
-          if (currentBase && currentBase.relative !== newBaseUrl) {
-            cheerio.prototype.options.xmlMode = false;
-            const rendered = nunjucks.renderString(cheerio.html(node.children), {
-              // This is to prevent the nunjuck call from converting {{hostBaseUrl}} to an empty string
-              // and let the hostBaseUrl value be injected later.
-              hostBaseUrl: '{{hostBaseUrl}}',
-              baseUrl: `{{hostBaseUrl}}/${newBaseUrl}`,
-            }, { path: filePath });
-            node.children = cheerio.parseHTML(rendered, true);
-            cheerio.prototype.options.xmlMode = true;
-          }
-        }
-      }
-      delete node.attribs[ATTRIB_INCLUDE_PATH];
-    }
+
+    // Rebase children elements
+    node.children.forEach(el => this._rebaseReference(el));
+
+    const includeSourceFile = node.attribs[ATTRIB_CWF];
+    const includedSourceFile = node.attribs[ATTRIB_INCLUDE_PATH];
     delete node.attribs[ATTRIB_CWF];
```

Comment:
Just wondering - should we delete these attributes here? It feels a little odd since they were added in `_preprocess` (at least, I think), and it isn't immediately intuitive that these attributes would be deleted in this function.

If it is necessary for us to delete it here, could we leave a comment explaining why we need to do this? And maybe add a comment in `_preprocess` to notify that this variable will be deleted here.

Reply:
The original function already deletes both

Reply:
It's not necessary to the best of my knowledge, but not having `<div cwf="...">` all over the output is a nice touch

Reply:
Ah noted I didn't notice that these outputs would be included in the output!